lib/msal-angular/src/msal.service.ts

```diff
@@ -173,11 +173,12 @@ export class MsalService extends UserAgentApplication {
             return null;
         }
 
-        if (this.msalConfig.framework && this.msalConfig.framework.protectedResourceMap) {
+        const frameworkProtectedResourceMap = this.msalConfig.framework && this.msalConfig.framework.protectedResourceMap;
```

Comment:
We should add a length check for this.msalConfig.framework.unprotectedResources too (Line 173 of this file).

Reply:
Ah yes, good catch, thanks!